dbuf-lsp/src/diagnostic_handler/code_lens.rs

```diff
@@ -0,0 +1,69 @@
+use tower_lsp::lsp_types::*;
+
+use crate::core::ast_access::{
+    ElaboratedAst, File, Loc, LocStringHelper, LocationHelpers, ParsedAst, Str,
+};
+use crate::core::navigator::Navigator;
+use crate::core::{ast_visitor::*, navigator};
+
+pub struct CodeLensProvider<'a> {
```

Comment:
why is this a separate type and not just a function which computes the code lenses?